File: Data Structures & Algorithms/three-integer-sum/submission-0.go
func threeSum(nums []int) [][]int {
	res := [][]int{}
	sort.Ints(nums)
	target:=0
	for i:=0;i<len(nums);i++{
		if i!=0 && nums[i]==nums[i-1]{continue}
		j := i+1
		k := len(nums)-1
	for j<k{
		if nums[i]+nums[j]+nums[k] == target {
			res = append(res, []int{nums[i],nums[j],nums[k]})
			j++
			k--
			for j<k && nums[j]==nums[j-1]{
				j++
			}
			for j<k && nums[k]==nums[k+1]{
				k--
			}
		}
		if nums[i]+nums[j]+nums[k] < target{
			j++
		}
		if nums[i]+nums[j]+nums[k] > target{
			k--
		}
		
	}
		
	}

return res
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.py
class Solution:
    def combinationSum2(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        nums.sort()
        def dfs(i, curr, currsum):
            if currsum>=target or i>=len(nums):
                if currsum == target:
                    res.append(curr.copy())
                return
            if currsum+nums[i]<=target:
                curr.append(nums[i])
                currsum+=nums[i]
                dfs(i+1, curr, currsum)
                curr.pop()
                currsum -= nums[i]
            
            x = 1
            for j in range(i+1, len(nums)):
                if nums[j]!=nums[i]:
                    break
                x+=1
            
            dfs(i+x, curr, currsum)

        dfs(0, [], 0)
        return res

            
            

